File: src/main/java/com/dude/dms/backend/repositories/RegexRuleRepository.java
package com.dude.dms.backend.repositories;

import com.dude.dms.backend.data.rules.RegexRule;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.List;

public interface RegexRuleRepository extends JpaRepository<RegexRule, Long> {

    List<RegexRule> findByActive(boolean active);
}

File: src/main/java/com/dude/dms/backend/repositories/RegexRuleRepository.kt
package com.dude.dms.backend.repositories

import com.dude.dms.backend.data.Tag
import com.dude.dms.backend.data.rules.RegexRule
import org.springframework.data.jpa.repository.JpaRepository

interface RegexRuleRepository : JpaRepository<RegexRule, Long> {

    fun findByTags(tag: Tag): Set<RegexRule>

    fun countByTags(tag: Tag): Long
}